### ISSUE TITLE
Upgrade the language filter used for the comments to be language-aware

### ISSUE THREAD

**Issue by Ascor8522** [LLK/scratch-www#6006]
Many users on the French-speaking section of the forums have been complaining recently about getting muted several time or their comments being flagged as inappropriate for no apparent reason.

Thing is, they don't know about the bad word in English with the same spelling. And the filter isn't helping. Many words judged as inappropriate in English are perfectly fine in other languages, like French in this case.

Below are a few examples:
https://scratch.mit.edu/discuss/topic/231655/?page=487#post-2900189
https://scratch.mit.edu/discuss/topic/319903/?page=28#post-4224660
https://scratch.mit.edu/discuss/topic/336305/
https://scratch.mit.edu/discuss/topic/337737/
https://scratch.mit.edu/discuss/topic/404922/
https://scratch.mit.edu/discuss/topic/416479/
https://scratch.mit.edu/discuss/topic/442971/
https://scratch.mit.edu/discuss/topic/461555/
https://scratch.mit.edu/discuss/topic/463055/
https://scratch.mit.edu/discuss/topic/483933/
https://scratch.mit.edu/discuss/topic/539754/

Note that those topics are only the ones I found or remembered about, there might be lots of other ones, in French but also in other languages.

This also applies in the forums when some words are censored for no "actual reason".

Also, you can't mention users with a username that might look like an inappropriate in English.

### Steps to Reproduce

1. Write a comment with any of the the following:
* `Je vais être en retard.` (I will be late)
* `Désolé du retard.` (Sorry for being late)
* `Je vais prendre ma douche` (I'll have a shower)
* `J'ai été pisser` (I went to the bathroom)
* `J'aime le fromage rapé` (I love grated cheese)
* `Tu dis quoi ?` (What are you saying?)
* `Salut Martin, C'est moi.` (Hello Martin, it's me)
* `Windows est installé sur le disque C:` (Windows is installed on the C drive)
* `Salut @Babichtouk comment ça va ?` (Hello Babichtouk, how are you doing?)
* `Le petit de la biche est le faon.` (The child of the hind is called the fawn)
2. Your comment isn't posted and you get an error message telling you to be respectful towards others.
3. Your account get muted for 5 to 10 minutes.

### Solution

Make the bad-word detector language aware.
Many libraries are available to detect what language a comment is in and are easy to use.
Then, the filter for that specific language should be applied.

Same for the forums.

**Comment by LankyBox01** [comment]
agreed

**Comment by AksharVivek** [comment]
Yes, and actual bad stuff don't get flagged at all like people are like "ayo wtf" but scratch didn't ban it since it was in abbreviated form, so Scratch team should update it so it flags abbreviated forms of cursing as well.
